homeassistant/components/ecobee/sensor.py

```diff
@@ -128,11 +128,16 @@ def unit_of_measurement(self):
     async def async_update(self):
         """Get the latest state of the sensor."""
         await self.data.update()
-        for sensor in self.data.ecobee.get_remote_sensors(self.index):
-            if sensor["name"] != self.sensor_name:
-                continue
-            for item in sensor["capability"]:
-                if item["type"] != self.type:
+        thermostat = self.data.ecobee.get_thermostat(self.index)
+        if not thermostat["runtime"]["connected"]:
+            # If the thermostat is disconnected, sensor data is stale
+            self._state = "unknown"
```

Comment:
I suggest setting `available` property instead. If we know that the thermostat is disconnected we know it's unavailable. `unknown` state means we don't know anything about the state.

Reply:
I played with the `available` property here, but wasn't sure it was really working previously. I'm testing it again and things look good.
Question, any reason not to use the `available` property for both the `sensor` and the `binary_sensor`? It looks to be working for both.

Reply:
Thanks. I tried out the `available` property previously but was having issues with it. I put the properties back in and now things seem to behaving better... Not sure what I messed up previously. Prepping an update now.